frontend/pages/MfaPage/MfaPage.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useState, useEffect } from "react";
+import { InjectedRouter } from "react-router";
+import { Params } from "react-router/lib/Router";
+
+import { AppContext } from "context/app";
+import { RoutingContext } from "context/routing";
+import paths from "router/paths";
+import local from "utilities/local";
+import configAPI from "services/entities/config";
+import sessionsAPI from "services/entities/sessions";
+
+import Button from "components/buttons/Button";
+import AuthenticationFormWrapper from "components/AuthenticationFormWrapper";
+import StackedWhiteBoxes from "components/StackedWhiteBoxes";
+import Spinner from "components/Spinner";
+
+interface IMfaPage {
+  router: InjectedRouter; // v3
+  params: Params;
+}
+
+const baseClass = "mfa-page";
+
+const MfaPage = ({ router, params }: IMfaPage) => {
+  const { token: mfaToken } = params;
+  const {
+    config,
+    currentUser,
+    setAvailableTeams,
+    setConfig,
+    setCurrentUser,
+    setCurrentTeam,
+  } = useContext(AppContext);
+  const { redirectLocation } = useContext(RoutingContext);
+  const [isExpired, setIsExpired] = useState(false);
+
+  const finishMFA = async () => {
+    const { DASHBOARD, RESET_PASSWORD, NO_ACCESS } = paths;
+
+    try {
+      const response = await sessionsAPI.finishMFA({ token: mfaToken });
+      const { user, available_teams, token } = response;
+
+      local.setItem("auth_token", token);
+
+      setCurrentUser(user);
+      setAvailableTeams(user, available_teams);
+      setCurrentTeam(undefined);
+
+      if (!user.global_role && user.teams.length === 0) {
+        router.push(NO_ACCESS);
+        return;
+      }
+      // Redirect to password reset page if user is forced to reset password.
+      // Any other requests will fail.
+      else if (user.force_password_reset) {
+        router.push(RESET_PASSWORD);
+        return;
+      } else if (config) {
+        router.push(redirectLocation || DASHBOARD);
+        return;
+      }
+
+      configAPI.loadAll().then((configResponse) => {
+        setConfig(configResponse);
+        router.push(redirectLocation || DASHBOARD);
+      });
+    } catch (response) {
```

Comment:
Will there be any other possible reasons for an error other than the MFA process being expired?

Reply:
Good question, @iansltx should we handle errors similar to `sessionsAPI.login`?